deeplearning4j/deeplearning4j-nlp-parent/deeplearning4j-nlp/src/main/java/org/deeplearning4j/iterator/BertIterator.java

```diff
@@ -0,0 +1,479 @@
+        }
+
+        //Determine output array length...
+        int outLength;
+        switch (lengthHandling){
+            case FIXED_LENGTH:
+                outLength = maxTokens;
+                break;
+            case ANY_LENGTH:
+                outLength = longestSeq;
+                break;
+            case CLIP_ONLY:
+                outLength = Math.min(maxTokens, longestSeq);
+                break;
+            default:
+                throw new RuntimeException("Not implemented length handling mode: " + lengthHandling);
+        }
+
+        int mb = tokenizedSentences.size();
+        int mbPadded = padMinibatches ? minibatchSize : mb;
+        int[][] outIdxs = new int[mbPadded][outLength];
+        int[][] outMask = new int[mbPadded][outLength];
+
+        for( int i=0; i<tokenizedSentences.size(); i++ ){
+            Pair<List<String>,String> p = tokenizedSentences.get(i);
+            List<String> t = p.getFirst();
+            for( int j=0; j<outLength && j<t.size(); j++ ){
+                Preconditions.checkState(vocabMap.containsKey(t.get(j)), "Unknown token encontered: token \"%s\" is not in vocabulary", t.get(j));
+                int idx = vocabMap.get(t.get(j));
+                outIdxs[i][j] = idx;
+                outMask[i][j] = 1;
+            }
+        }
+
+        //Create actual arrays. Indices, mask, and optional segment ID
+        INDArray outIdxsArr = Nd4j.createFromArray(outIdxs);
+        INDArray outMaskArr = Nd4j.createFromArray(outMask);
+        INDArray outSegmentIdArr;
+        INDArray[] f;
+        INDArray[] fm;
+        if(featureArrays == FeatureArrays.INDICES_MASK_SEGMENTID){
+            //For now: always segment index 0 (only single s sequence input supported)
+            outSegmentIdArr = Nd4j.zeros(DataType.INT, mbPadded, outLength);
+            f = new INDArray[]{outIdxsArr, outSegmentIdArr};
+            fm = new INDArray[]{outMaskArr, null};
+        } else {
+            f = new INDArray[]{outIdxsArr};
+            fm = new INDArray[]{outMaskArr};
+        }
+
+        INDArray[] l = new INDArray[1];
+        INDArray[] lm;
+        if(task == Task.SEQ_CLASSIFICATION){
+            //Sequence classification task: output is 2d, one-hot, shape [minibatch, numClasses]
+            int numClasses;
+            int[] classLabels = new int[mbPadded];
+            if(sentenceProvider != null){
+                numClasses = sentenceProvider.numLabelClasses();
+                List<String> labels = sentenceProvider.allLabels();
+                for(int i=0; i<mb; i++ ){
+                    String lbl = tokenizedSentences.get(i).getRight();
+                    classLabels[i] = labels.indexOf(lbl);
+                    Preconditions.checkState(classLabels[i] >= 0, "Provided label \"%s\" for sentence does not exist in set of classes/categories", lbl);
+                }
+            } else {
+                throw new RuntimeException();
```

Comment:
Can this case even happen? Earlier in the method you are already throwing an exception if sentencProvider is null

Reply:
Not yet, but it's a guard against future incomplete changes.